classes/Api/ThankYous.php

```diff
@@ -0,0 +1,809 @@
+
+			$thank_you->SetComment($thank_yous_comments[$id]);
+		}
+		$this->LoadThankYousCommentsTotalComments($thank_yous_comments);
+	}
+
+	/**
+	 * Sets Thank Yous Comments' Total Comments.
+	 *
+	 * @param Comments\CommentableThankYou[] $thank_yous_comments
+	 * @return array
+	 */
+	public function LoadThankYousCommentsTotalComments(array $thank_yous_comments): array
+	{
+		foreach ($thank_yous_comments as $thank_you_comments)
+		{
+			if (!$this->comments_factory->IsCommentInstance($thank_you_comments))
+			{
+				throw new InvalidArgumentException("Failed to Load Thank Yous' Comments Total Comments, one ore more entities provided is not a Thank You Comments");
+			}
+
+			$comments_total_count = (int) $this->comments_repository->GetCommentsCount($thank_you_comments);
+			$thank_you_comments->SetTotalComments($comments_total_count);
+		}
+
+		return $thank_yous_comments;
+	}
+
+	/**
+	 * @param ThankYou $thank_you
+	 * @return string
+	 */
+	public function GetThankYouUrl(ThankYou $thank_you)
+	{
+		$id = $thank_you->GetId();
+
+		if (!isset($id))
+		{
+			throw new InvalidArgumentException("Failed to Get Thank You's URL, Thank You's ID is unknown");
+		}
+
+		return $this->GetThankYouUrlById($id);
+	}
+
+	/**
+	 * @param int $id
+	 * @return string
+	 */
+	public function GetThankYouUrlById(int $id)
+	{
+		return $this->utility->GetThankYouUrl($id);
+	}
+
+	/**
+	 * @param User|int  $author
+	 * @param string    $description
+	 * @param Date|null $date_created
+	 * @return ThankYou
+	 * @throws ThankYouAuthor - If the Author could not be loaded.
+	 */
+	public function Create($author, string $description, ?Date $date_created = null)
+	{
+		return $this->thank_yous_repository->Create($author, $description, $date_created);
+	}
+
+	/**
+	 * @param int $o_class
+	 * @param int $id
+	 * @return Thankable
+	 * @throws ThankYouOClass - If the Owner Class given is not supported.
+	 */
+	public function CreateThankableFromOClass(int $o_class, int $id)
+	{
+		return $this->CreateThankablesFromOClasses([['oclass' => $o_class, 'id' => $id]])[0];
+	}
+
+	/**
+	 * Takes an array of arrays in the format ['oclass' => int, 'id' => int]
+	 * Returns an array of Thanked Objects, retaining indexing.
+	 *
+	 * @param array $oclasses
+	 * @return Thankable[]
+	 * @throws ThankYouOClass - If one or more of the Owner Classes given is not supported.
+	 */
+	public function CreateThankablesFromOClasses(array $oclasses)
+	{
+		return $this->thank_yous_repository->CreateThankablesFromOClasses($oclasses);
+	}
+
+	/**
+	 * Save a Thank You to the Repository and generate an Audit. If the Thank You doesn't have an ID, one will be set.
+	 *
+	 * @param ThankYou $thank_you
+	 * @throws ThankYouNotFound - If the Thank You could not be found in the Repository.
+	 * @throws TagNotFound If one or more of the Thank You's Tags could not be found in the Repository.
+	 */
+	public function Save(ThankYou $thank_you)
+	{
+		$new = ($thank_you->GetId() === null) ? true : false;
```

Comment:
```suggestion
		$new = empty($thank_you->GetId());
```

Reply:
I disagree with this, as it's not quite the same check. `empty` will return `TRUE` if `GetId` returns `0`, which is completely valid with regards to how the IDs are handled by the ThankYous library. Whilst it would indeed be a strange value to return, I don't agree that it's worth this change to make an approximation of the check we actually intended.

Originally I remember wanting to use `isset`, but couldn't as that can't be used on a method. A work around to this could be
```
$new = $thank_you->GetId();
$new = isset($new);
```
What are your thoughts?

Reply:
If you definitely want to explicitly check for `null`, then this is fine. Ultimately I just wanted to remove the redundant `? true : false` ternary expression, and across our codebase IDs can often default to `0`. Perhaps not in this case, though!

```suggestion
		$new = $thank_you->GetId() === null;
```

Reply:
Derh!!
Fixed in e9cef15fae82fcdd13af03ddb1fcbf033da81d48